syntax = "proto3";

import "google/protobuf/empty.proto";

message AppendRequest {
  string data = 1;
}

message AppendReply {
  uint64 id = 1;
}

message ReadRequest {
  uint64 id = 1;
}

message ReadReply {
  string data = 1;
}

service QueueService {
  rpc Append(AppendRequest) returns (AppendReply) {}
  rpc Read(ReadRequest) returns (ReadReply) {}
}

message EchoRequest {
  string msg = 1;
}

message EchoReply {
  string msg = 1;
}

service EchoService {
  rpc Echo(EchoRequest) returns (EchoReply) {}
}
